src/pages/index.js

```diff
@@ -42,54 +33,175 @@ export function handleImageClick(name, link) {
 
-// enablePopupWithForm on newCard
-const newCardPopup = new PopupWithForm(
-  "#profile-add-modal",
-  handleNewCardSubmit
-);
-newCardPopup.setEventListeners();
+// Render user info and cards on page load
+Promise.all([api.getUser(), api.getInitialCards()]).then(([data, cards]) => {
+  userInfo.setUserInfo({ name: data.name, about: data.about });
+  userInfo.setUserAvatar(data.avatar, data.name);
+  cardList = new Section(
+    { data: cards, renderer: renderCard },
+    ".cards__list",
+    handleImageClick,
+    handleDeleteClick,
+    handleLikeClick
+  );
+  cardList.renderItems();
+});
 
-// enable PopupWithImage on cards
-const cardPopup = new PopupWithImage("#card-modal");
-cardPopup.setEventListeners();
+// enabling validation by calling enableValidation()
+const editFormValidator = new FormValidator(config, profileEditForm);
+editFormValidator.enableValidation();
+
+const addFormValidator = new FormValidator(config, addCardForm);
+addFormValidator.enableValidation();
+
+const avatarFormValidator = new FormValidator(config, avatarForm);
+avatarFormValidator.enableValidation();
+
+// Editing Profile Interaction
+// Edit Profile Information
+const editProfilePopup = new PopupWithForm("#profile-edit-modal", (data) => {
+  editProfilePopup.setSubmitText(true);
+  console.log(data);
+  api
+    .updateProfile(data)
+    .then((data) => {
+      userInfo.setUserInfo(data);
+      editProfilePopup.close();
+    })
+    .finally(() => editProfilePopup.setSubmitText(false));
+});
+editProfilePopup.setEventListeners();
 
 // On click event for editProfile button
 profileEditButton.addEventListener("click", () => {
   editProfilePopup.open();
   const userData = userInfo.getUserInfo();
   inputName.value = userData.name;
-  inputDescription.value = userData.job;
+  inputDescription.value = userData.about;
+  let newInfo = { name: inputName.value, about: inputDescription.value };
+  userInfo.setUserInfo(newInfo);
 });
 
+// New Card Interaction
+// Add new card
+const newCardPopup = new PopupWithForm(
+  "#profile-add-modal",
+  ({ title, image }) => {
+    newCardPopup.setSubmitText(true);
+    api
+      .addCard(title, image)
+      .then((card) => {
+        newCard = createCard({
+          name: card.name,
+          link: card.link,
+          _id: card._id,
+          isLiked: false,
+        });
+        cardList.addItem(newCard);
+      })
+      .catch((err) => console.log(err))
+      .finally(() => newCardPopup.setSubmitText(false));
+  }
+);
+newCardPopup.setEventListeners();
+
 // On click event for addCard button
 profileAddButton.addEventListener("click", () => {
   addFormValidator.disableButton();
   newCardPopup.open();
 });
+
+// Profile Picture Interaction
+// Update Profile Picture
+const updateAvatarPopup = new PopupWithForm("#update-avatar-modal", (image) => {
+  updateAvatarPopup.setSubmitText(true);
+  api
+    .updateProfilePicture(image.image)
+    .then((userData) => {
+      userInfo.setUserAvatar(userData.avatar);
```

Comment:
Needs correction: `TypeError: Cannot read properties of undefined (reading 'avatar')
    at index.js:171:39`

Reply:
I'm also not able to replicate this error. How did this show up for you?

Reply:
I can't replicate it, seems like you've fixed it by fixing something else